Сделал иконку приложения

PanBuz opened the pull request.

PanBuz committed:
Сделал иконку приложения

PanBuz committed:
спринт 23

PanBuz committed:
Удалил лишнее

Nick-review commented on app/src/main/java/com/example/playlistmaker/main/MainActivity.kt:
```diff
@@ -29,7 +28,8 @@ class MainActivity : AppCompatActivity() {
 
         navController.addOnDestinationChangedListener {_, nd, _ ->
```
может какое-то более осознанное название? что такое nd совсем непонятно)

PanBuz replied:
поменял)

PanBuz committed:
убрал проверку миграции базы данных

PanBuz committed:
изменил нейминг

PanBuz committed:
подправил дизайн картинок

PanBuz committed:
Добавил логику сохранения картинки при редактировании плейлиста

PanBuz committed:
При отсутствие описания плейлиста, отсутсвует отступ между названием трека и колличеством

PanBuz committed:
"Создать плейлист. Нажать на "Редактирование плейлиста". Теперь в полях ввода вместо "название" и "описание" отображаются
 введенные туда данные при создании. Названия полей должны всегда оставаться неизменными." исправил

PanBuz committed:
пока не удалось победить наличие картинок в выпадающем списке плейлистов при добавление песни((

PanBuz committed:
теперь картинки в выпадающем списке отображаются

Nick-review commented on app/src/main/res/values/dimens.xml:
```diff
@@ -30,5 +30,6 @@
     <dimen name="text_22sp">22sp</dimen>
     <dimen name="text_19sp">19sp</dimen>
     <dimen name="text_14sp">14sp</dimen>
+    <dimen name="text_11sp">11sp</dimen>
```
Как правило, размеры лучше брать кратные 4, или на крайний случай 2. Хоть часто в дизайне и указываются нечетные числа

Nick-review commented on app/src/main/res/navigation/main_navigation_graph.xml:
```diff
@@ -94,4 +97,42 @@
+        <action
+            android:id="@+id/action_updatePlaylistFragment_to_newPlaylistFragment"
+            app:destination="@id/newPlaylistFragment" />
+        <argument
```
Это дело вкуса конечно, но как по мне лучше и нагляднее работать с аргументами из кода и класть в аргументы в ручную, это нагляднее:)
P.s. ни разу не видел на настоящих проектах такого способа, возможно просто  непривычно)

Nick-review commented on app/src/main/res/layout/layout_playlist.xml:
```diff
@@ -30,7 +30,7 @@
         android:textColor="@color/text_color"
         android:textFontWeight="400"
-        android:textSize="12dp"
+        android:textSize="12sp"
```
это в dimens.xml Лучше спрятать

Nick-review commented on app/src/main/res/layout/fragment_display_playlist.xml:
```diff
@@ -0,0 +1,298 @@
+            android:fontFamily="@font/ys_display_regular"
+            android:textColor="@color/text_color"
+            android:textFontWeight="400"
+            android:textSize="16sp"
```
это в dimens.xml Лучше спрятать тут и во всём файле

PanBuz replied:
Обещаю, сделать это позже)

Nick-review commented on app/src/main/java/com/example/playlistmaker/utils/Converters.kt:
```diff
@@ -0,0 +1,42 @@
+
+class Converters (val context:Context){
+
+    val context1 = android.R.string()
```
Может удалить если не используется?

PanBuz replied:
Да да, удалил)

Nick-review commented on app/src/main/java/com/example/playlistmaker/utils/Converters.kt:
```diff
@@ -0,0 +1,42 @@
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Converters (val context:Context){
```
Можно сделать context private

PanBuz replied:
Сделал

Nick-review commented on app/src/main/java/com/example/playlistmaker/player/ui/PlayerFragment.kt:
```diff
@@ -100,6 +109,7 @@ class PlayerFragment : Fragment() {
         when (playlistState) {
             PlaylistState.Empty -> {}
             is PlaylistState.Playlists -> {
+               // adapter.imagePath = newViewModel.imagePath()
```
Удалить если не используется?

PanBuz replied:
Удалил

Nick-review commented on app/src/main/java/com/example/playlistmaker/player/ui/PlayerAdapter.kt:
```diff
@@ -2,14 +2,21 @@ package com.example.playlistmaker.player.ui
     var playlists = arrayListOf<Playlist>()
     var clickListener: ((Playlist) -> Unit)? = null
-
+    lateinit var imagePath : String
```
Если изменить порядок инициализации в player fragment, то в месте вызова это упадёт. Лучше через нулабельное поле сделать

PanBuz replied:
Спасибо, сделал.

Nick-review commented on app/src/main/java/com/example/playlistmaker/player/ui/PlayerAdapter.kt:
```diff
@@ -2,14 +2,21 @@ package com.example.playlistmaker.player.ui
+
+
     var playlists = arrayListOf<Playlist>()
     var clickListener: ((Playlist) -> Unit)? = null
```
И в целом тут нарушается правило инкапсуляции, листенер можно прокидывать в конструкторе, а для остальный свойств сделать методы

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/updatePlaylist/UpdatePlaylistViewModel.kt:
```diff
@@ -0,0 +1,61 @@
+    fun initialization() {
+        // Второй вариант передачи данных плэйлиста из DisplayPlaylistFragment
+        val actualPlaylist = DisplayPlaylistFragment.actualPlaylist
+        _updateLiveData.postValue(actualPlaylist !!)
```
Лучше не использовать !!, а обрабатывать такие кейсы с нулами

PanBuz replied:
Переделал

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/updatePlaylist/UpdatePlaylistViewModel.kt:
```diff
@@ -0,0 +1,61 @@
+
+    fun initialization() {
+        // Второй вариант передачи данных плэйлиста из DisplayPlaylistFragment
+        val actualPlaylist = DisplayPlaylistFragment.actualPlaylist
```
Вообще синглтон как антипатерн считается и в данном случае это ещё ничего, но если одна переменная меняется из многих мест и кто-то ещё считывает, то это рано или поздно приведет к неконсистентности
Получше будет, если акутальный плейлист будет храниться где-нибудь в шине данных, например, в репозитории и из нужных мест меняться и читать с него актуальные данные

PanBuz replied:
Пока оставил так, но буду стараться переделать.

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/updatePlaylist/UpdatePlaylistViewModel.kt:
```diff
@@ -0,0 +1,61 @@
+    newPlaylistInteractor: NewPlaylistInteractor
+) : NewPlaylistViewModel(interactor, newPlaylistInteractor) {
+    private var _updateLiveData = MutableLiveData<Playlist>()
+    val updateLiveData: LiveData<Playlist> = _updateLiveData
```
update и updateLiveData не самые понятные нейминги, лучше писать что конкретно они обновляют (updateState, updateSongList и т.п.) и в названии переменной не имеет смысла писать что это LiveData, это видно из типа переменной

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/updatePlaylist/UpdatePlaylistFragment.kt:
```diff
@@ -0,0 +1,137 @@
+    }
+
+    private fun checkChanges() : Boolean {
+        if  (binding.ietNamePl.text!!.isEmpty())   { return false }
```
тут тоже лучше отказаться от !! и обрабатывать нулы

PanBuz replied:
Переделал

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/updatePlaylist/UpdatePlaylistFragment.kt:
```diff
@@ -0,0 +1,137 @@
+    override val viewModel by viewModel<UpdatePlaylistViewModel>()
+    var updatePlaylist : Playlist = Playlist(0, "", "", "", arrayListOf(),0,0)
+    //var selectedUri = viewModel.selectedUri
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```
Всю эту функцию лучше разбить на смысловые блоки и функции поменьше, так будет понятнее читаться

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/newPlaylist/NewPlaylistViewModel.kt:
```diff
@@ -13,13 +14,13 @@ import com.example.playlistmaker.mediateka.domain.newPlaylist.NewPlaylistInterac
-    private val interactor: PlaylistInteractor,
-    private val newPlaylistInteractor: NewPlaylistInteractor,
+open class NewPlaylistViewModel(
+    val interactor: PlaylistInteractor,
```
их лучше сделать Internal, чтобы только наследники имели к ним доступ

PanBuz replied:
Изменил

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/newPlaylist/NewPlaylistViewModel.kt:
```diff
@@ -13,13 +14,13 @@ import com.example.playlistmaker.mediateka.domain.newPlaylist.NewPlaylistInterac
 
     private lateinit var pickMedia: ActivityResultLauncher<PickVisualMediaRequest>
-    private var selectedUri: Uri? = null
+    var selectedUri: Uri? = null
```
а тут private

PanBuz replied:
изменил

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/newPlaylist/NewPlaylistViewModel.kt:
```diff
@@ -13,13 +14,13 @@ import com.example.playlistmaker.mediateka.domain.newPlaylist.NewPlaylistInterac
-class NewPlaylistViewModel(
-    private val interactor: PlaylistInteractor,
-    private val newPlaylistInteractor: NewPlaylistInteractor,
+open class NewPlaylistViewModel(
```
И в целом такой функционал лучше выносить в base view model, а от неё уже наследоваться

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/ui/displayPlaylist/DisplayPlaylistViewModel.kt:
```diff
@@ -0,0 +1,76 @@
+    private var _playlistLiveData = MutableLiveData<Playlist>()
+    val playlistLiveData: LiveData<Playlist> = _playlistLiveData
+    private var _liveData = MutableLiveData<PlaylistState>()
+    val liveData: LiveData<PlaylistState> = _liveData
```
Тут аналогично с неймингами

Nick-review commented on app/src/main/java/com/example/playlistmaker/mediateka/data/dao/FavoriteDao.kt:
```diff
@@ -15,16 +15,25 @@ interface FavoriteDao {
+    suspend fun insertTrack(track: FavoriteTraksEntity)
+
+    //@Query("SELECT * FROM favoritetraks_table")
+    //suspend fun getTraks(): List<FavoriteTraksEntity>
```
Удалить неиспользуемое?

PanBuz replied:
да да

PanBuz committed:
Не сохраняется новая картинка. Теперь при входе в редактирование приложение падает (вызов конструктора фрагмента вызвал исключение)

PanBuz committed:
Нашел проблему, решил.

PanBuz committed:
После принятия, закоментированное уберу.

PanBuz committed:
Доработка null

PanBuz committed:
Доработка замечаний

PanBuz committed:
Доработка замечаний

PanBuz committed:
Добавил комент
